stellar-contract-env-host/src/test.rs

```diff
@@ -129,3 +129,226 @@ fn empty_vec_back() {
     let obj = host.to_host_obj(&scobj).unwrap();
     host.vec_back(*obj.as_ref()).unwrap();
 }
+
+#[test]
+fn vec_put_and_get() {
+    let mut host = Host::default();
+    let scvec: ScVec = vec![ScVal::U32(1), ScVal::U32(2), ScVal::U32(3)]
+        .try_into()
+        .unwrap();
+    let scobj = ScObject::Vec(scvec);
+    let obj = host.to_host_obj(&scobj).unwrap();
+    let i: RawVal = 1_u32.into();
+    let obj1 = host.vec_put(*obj.as_ref(), i, 9_u32.into()).unwrap();
```

Comment:
Since you're not really wanting a ref to obj, using as_ref then copying is a bit of a roundabout way to do the conversion. We have `to_tagged()` that will convert the obj into a `TaggedVal<TagObject>`, but I can definitely see it is super not obvious that `to_tagged()` would get you what you want since we use the alias `Object` everywhere instead `TaggedVal<TagObject>`. So I just added `to_object()` in https://github.com/stellar/rs-stellar-contract-env/pull/84 that does the same thing with a better name, and I think it will fit your intent clearer here:
```suggestion
    let obj1 = host.vec_put(obj.to_object(), i, 9_u32.into()).unwrap();
```

Reply:
Ah, I missed the `to_tagged()`, you are right. Thanks for adding the `to_object()`.